api.go

```diff
@@ -0,0 +1,45 @@
+package slack
+
+import (
+	"context"
+	"errors"
+	"net/url"
+)
+
+// APIResponseFull contains our API response from the api.test endpoint
+type APIResponseFull struct {
+	SlackResponse
+	Args map[string]string `json:"args,omitempty"`
+}
+
+func apiRequest(ctx context.Context, client httpClient, path string, values url.Values, debug debug) (*APIResponseFull, error) {
+	response := &APIResponseFull{}
+	err := postSlackMethod(ctx, client, path, values, response, debug)
+	if err != nil {
+		return nil, err
+	}
+	if !response.Ok {
+		return nil, errors.New(response.Error)
+	}
+	return response, nil
+}
+
+// GetApiTest will test the API by hitting the api.test endpoint, and
+// returning that response. It is a good way to make sure you're authenticated,
+// and talking to slack.
+func (api *Client) GetApiTest() (*APIResponseFull, error) {
```

Comment:
lets be a little more idiomatic.

```go
func (api *Client) Test() (err error) {
   response := SlackResponse{}
   if err = postSlackMethod(ctx, client, path, values, &response, api.debug); err != nil {
      return err
   }

    return response.Err()
}
```

if we were to expose the full api becomes a little more complicated.
```go
func (api *Client) Test(paylod json.RawMessage) (resp json.RawMessage, err error) {
   // ... again more complicated as you'll need to make the API call, get the raw response and then parse it to check for errors.
}
```